### ISSUE TITLE
cucumber2: incorrect selection of scenarios to execute

### ISSUE THREAD

**Issue by akostadinov** [cucumber/cucumber-ruby#778]
Hello, I'm running cucumber like:
`cucumber example.feature:lineX example.feature:lineY`

With cucumber 1.3 I see scenarios executed in the order specified once. With cucumber 2 beta3 I see that first it runs the scenario with lower line number first and then it runs the same two scenarios a second time.

The order of execution is a problem because we are using an external test management system to synchronize scenario execution across multiple runners. But it is not ideal for the purpose and when having scenarios in same order on all runners, there are duplicate runs. To avoid that we reorder scenarios randomly and collisions are unlikely. With cucumber reordering execution, this may not work well.
Executing scenarios multiple times just doesn't make sense to me.

There is one more thing that I'm not sure is a bug but is a difference between 1.3 and 2.x. When specifying a line number of a scenario step, then 1.3 would execute the whole scenario. With 2.0 nothing gets executed.

To summarize:
- [ ] order of scenarios is not preserved
- [x] scenarios executed more than once
- [x] 2.x would not execute a scenario specified by the line number of a `step` from that scenario
